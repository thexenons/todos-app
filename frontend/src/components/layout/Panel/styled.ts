import styled from "styled-components";

export const PanelWrapper = styled.article`
	background-color: ${({ theme }) => theme.colors.background.paper};
	border-radius: ${({ theme }) => theme.spacing(2)};
	min-width: 300px;
	max-width: 600px;
	${({ theme }) => theme.mixins.flex({ direction: "column" })}
`;

export const PanelContent = styled.section`
	${({ theme }) => theme.mixins.padding({ all: 2 })}
	flex: 1 1 auto;
`;

export const PanelHeader = styled.header`
	${({ theme }) => theme.mixins.padding({ x: 2, top: 2 })}
`;

export const PanelFooter = styled.footer`
	${({ theme }) => theme.mixins.padding({ x: 2, bottom: 2 })}
`;
