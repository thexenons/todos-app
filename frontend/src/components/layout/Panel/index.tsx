import * as S from "./styled";

const Panel = S.PanelWrapper as typeof S.PanelWrapper & {
	Header: typeof S.PanelHeader;
	Content: typeof S.PanelContent;
	Footer: typeof S.PanelFooter;
};

Panel.Header = S.PanelHeader;
Panel.Content = S.PanelContent;
Panel.Footer = S.PanelFooter;

export default Panel;
