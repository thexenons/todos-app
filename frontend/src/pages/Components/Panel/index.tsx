import { FC } from "react";

import Panel from "../../../components/layout/Panel";
import * as SS from "../styled";

const PanelComponentsPage: FC = () => (
	<SS.ComponentsListWrapper>
		<h1>Panel</h1>
		<SS.ComponentsRowWrapper>
			<Panel>
				<Panel.Content>Children</Panel.Content>
			</Panel>
			<Panel>
				<Panel.Header>Header</Panel.Header>
				<Panel.Content>Children</Panel.Content>
			</Panel>
			<Panel>
				<Panel.Content>Children</Panel.Content>
				<Panel.Footer>Footer</Panel.Footer>
			</Panel>
			<Panel>
				<Panel.Header>Header</Panel.Header>
				<Panel.Content>Children</Panel.Content>
				<Panel.Footer>Footer</Panel.Footer>
			</Panel>
		</SS.ComponentsRowWrapper>
	</SS.ComponentsListWrapper>
);

export default PanelComponentsPage;
