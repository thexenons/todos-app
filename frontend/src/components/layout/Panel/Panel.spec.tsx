import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import Panel from ".";

describe("Panel", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<Panel />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
