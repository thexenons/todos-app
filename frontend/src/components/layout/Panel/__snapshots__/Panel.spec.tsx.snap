// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Panel > Render 1`] = `
<div>
  <article
    class="styled__PanelWrapper-sc-hifdw9-0 ekCCrh"
  />
</div>
`;
